File: VIPVerify/src/com/vip/vipverify/net/NetSocketData.java
package com.vip.vipverify.net;

import java.io.Serializable;

public abstract class NetSocketData implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	protected abstract byte[] prepare_data();

	final public int be_send(SocketProxy socket_proxy) {
		int nret = -1;
		byte[] send_byte = prepare_data();
		if (socket_proxy != null && send_byte != null) {
			nret = socket_proxy.send_data(send_byte, send_byte.length);
		}
		return nret;
	}
}

File: VerifyServer/NetSocketData.cpp
#include "NetSocketData.h"
#include "NetSocketProxy.h"

CNetSocketData::CNetSocketData(void)
{
}


CNetSocketData::~CNetSocketData(void)
{
}

int CNetSocketData::BeSend(CNetSocketProxy *pProxy, CEncrypter *pEncrypter /* = 0 */){
	int nret = -1;

	unsigned char send_data[1024] = {0};
	int len = _prepare_data(send_data, 1023);

	if (pProxy!=0 && len>0)
	{
		pProxy->SendData(send_data, len);
	}

	return nret;
}
